gap/SnAnBB.gi

```diff
@@ -18,100 +18,917 @@
 ##  The code is based upon the algorithm presented in the paper [BLGN+03].
 ##
 ##
-# DeclareInfoClass( "InfoRecSnAn" );
-# SetInfoLevel( InfoRecSnAn, 0 );
-#
-# # The following are used for comparisons:
-#
-# # FIXME: Get rid of these globals.
-# # However, it seems to me this whole file is dead code anyway..
-# RecSnAnIsOne := IsOne;
-# RecSnAnEq := EQ;
-#
-# ## is <r,s> in S_n
-#
-# SatisfiesSnPresentation := function( n, r, s )
-#
-#     local j, t;
+BindGlobal("SatisfiesSnPresentation",
+function(ri, n, r, s)
+    local j, t;
+    if not(isone(ri)((r * s)^(n-1))) then
+        return false;
+    fi;
+    j := 2;
+    t := r;
+    while j <= n/2 do
+        t := t * r;
+        if not isone(ri)(Comm(s,t)^2) then
+            return false;
+        fi;
+        j := j + 1;
+    od;
+    return true;
+end);
+
+BindGlobal("SatisfiesAnPresentation",
+function(ri, s, t, n)
```

Comment:
A bit confusing that `n` is in a different position here than for `SatisfiesSnPresentation`